### PR TITLE
Remaining Python 3 Compatibility

### PR DESCRIPTION
**Pull request by peternewman**
I think this fixes the remaining Python 3 compatibility.

I've not really given most of it any serious testing, so just getting some people to try using it would be a good start!

I'd appreciate any input or comments, whether general or from testing from anyone please.

CC @sdbbs @brucelowekamp @RenZ0 @danielskeenan @mikacousin

### CODE REVIEW

File: tools/ola_mon/ola_mon.py [hunk 1]
```diff
@@ -58,7 +67,10 @@ def _FetchDebug(self):
     try:
       response = connection.getresponse()
       if response.status == 200:
-        return response.read()
+        if sys.version >= '3.2':
```
Comment (peternewman) [inline tools/ola_mon/ola_mon.py:70]: As per #1755
```suggestion
        if sys.version_info >= (3, 2):
```

**Review by peternewman — COMMENTED** [review]
Todo to fix more of #1755 !

File: tools/ola_mon/ola_mon.py (hunk 1 above)
Comment (mikacousin) [reply tools/ola_mon/ola_mon.py:70]: Good catch !

**Comment by mikacousin** [comment]
With adding PR #1755 , `make check` success with python 3.10. 

Maybe, we can use `threading.current_thread()` instead of `threading.currentThread()` as is marked as deprecated (line 270 in ClientWrapper.py).
As I understand, it should work with python 2.6+. What is minimal python version ola support ?

**Comment by peternewman** [comment]
> With adding PR #1757 , `make check` success with python 3.10.

Excellent thanks, merged, as you've probably seen.

> Maybe, we can use `threading.current_thread()` instead of `threading.currentThread()` as is marked as deprecated (line 270 in ClientWrapper.py). As I understand, it should work with python 2.6+. What is minimal python version ola support ?

Yeah that sounds fine. My vague theory is to continue to target Python 2.6+ for 0.10 and master, and then master+1 can drop all of that (given the current codebase should support it, and at least one person was still using some ancient OS on an embedded system, and it seems a shame to needlessly throw away that work). My laptop isn't much newer than that, although it really should be...

**Comment by peternewman** [comment]
> > Maybe, we can use `threading.current_thread()` instead of `threading.currentThread()` as is marked as deprecated (line 270 in ClientWrapper.py). As I understand, it should work with python 2.6+. What is minimal python version ola support ?
> 
> Yeah that sounds fine.

Done by @mikacousin in #1758 .

File: tools/rdm/ResponderTestTest.py [hunk 2]
```diff
@@ -54,5 +54,62 @@ def testCmp(self):
...
+    self.assertEqual('%s' % rtf._EscapeData(['a']), "['a']")
+    self.assertEqual('%s' % rtf._EscapeData(["foo", 'a']), "['foo', 'a']")
+    self.assertEqual('%s' % rtf._EscapeData(['caf\xe9']), "['caf\\\\xe9']")
+    self.assertEqual('%s' % rtf._EscapeData(["foo", u'foo\u2014bar']),
+                     "['foo', 'foo\\\\u2014bar']")
+
+    self.assertEqual('%s' % rtf._EscapeData({"a": 0}), "{'a': 0}")
+    # The tests below that are commented out are non-deterministic on Python 3
+    # Might be able to get round it with a sort or something? We don't
+    # actually care about the order...
```
[the remark this answers is not in the document]
Comment (peternewman) [reply tools/rdm/ResponderTestTest.py:96]: Done in https://github.com/OpenLightingProject/ola/pull/1687/commits/d26d041132c33416c19194c2146b51153a0230f6 which will get merged in due course.

**Pull request merged**